### PR TITLE
v0.1.0 candidate - support for EdDSA, Schnorr & fix dependencies

### PR DESCRIPTION
**Pull request by oleiba**
- Support Schnorr (as in [Zilliqa](https://github.com/Zilliqa/Zilliqa-JavaScript-Library/tree/dev/packages/zilliqa-js-crypto) implementation)
- Support Ed25519
- Relevant tests added to `integration-tests` (all pass)
- `bitcoin` dependency upgraded to `0.20.0`
- Dependencies to rely on fixed tagged versions

This PR extends and replaces #30 .

### CODE REVIEW

**Comment by gbenattar** [comment]
I see no changes between this and two-party-eddsa. Is that expected? Is the purpose of this change to tag master?

Thanks!

**Comment by oleiba** [comment]
Here's the difference between `v0.1.0` and `two-party-eddsa`:
https://github.com/KZen-networks/gotham-city/compare/two-party-eddsa...v0.1.0

Yes, generally these added commits upgrade some dependencies (such as `bitcoin`) and adapt the source code to support the newer versions; and use tagged versions for dependencies.

**Comment by omershlo** [comment]
why ecdsa is used with branch = "alt-curv-v0.2.0" and not master? @oleiba

**Comment by oleiba** [comment]
Good catch.
The reason is that `multi-party-eddsa` uses curv's feature `ec_ed25519`, in contrast to the other packages which use the feature `ec_secp256k1`. 
These two features can not be used with the _same_ `curv` git repository.

My solution was to fork `curv` into another account (`oleiba`, until a better alternative is found) and have `multi-party-eddsa` to use that.

Because it is preferred that the master branch of `multi-party-eddsa` will not use a forked version of `curv` on master, I use a branch for it.

**Comment by omershlo** [comment]
I approve except for the double use of Curv repository. 
My main concern is that we will have to update oleiba/curv while you are on vacation.

**Comment by gbenattar** [comment]
What is preventing us to send a PR with Oded's change in curv for curv master?

**Comment by omershlo** [comment]
I would like to explore other options as @gbenattar suggested

**Comment by jeremyfelder** [comment]
Approved @gbenattar changes to DockerFile and tests

**Comment by gbenattar** [comment]
I am putting that PR on hold until @omershlo and @oleiba will have a a discussion about the dependency graph.

**Comment by oleiba** [comment]
After our discussion, changed `multi-party-eddsa` dependency to use tag `curv-replica@v0.2.0`.
It used htttps://github.com/KZen-networks/curv-replica, a replica of https://github.com/KZen-networks/curv, and the specific tag points to same commit as tag `v0.2.0` in the original repository.

**Comment by omershlo** [comment]
@oleiba have you tried to use `package` in Cargo.toml instead of curv2: 
https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml

something like : 
curv1 = { git = "https://github.com/KZen-networks/curv", version = "0.2.0", features = ['ec_secp256k1'], package = "curv" }
curv2 = { git = "https://github.com/KZen-networks/curv", version = "0.2.0", features = ['ec_ed25519'], package = "curv" }

**Comment by oleiba** [comment]
I have updated the Cargo.toml files with using `multi-party-eddsa` with tag `v0.2.1` - this tag is used in https://github.com/multi-party-eddsa **master** branch, and uses the same commit of `curv` as the direct dependency, but only with a different name (`v0.2.0-ed25519` instead of `v0.2.0`).

This removes the need of an additional `curv` repository, and I think it solves the issue nicely.

**Comment by gbenattar** [comment]
@omershlo is it good to go on your side?

**Comment by omershlo** [comment]
still working on it.

File: gotham-client/Cargo.toml [hunk 1]
```diff
@@ -21,32 +24,42 @@ log = "0.4"
 time = "*"
 clap = { version = "2.32", features = ["yaml"] }
 reqwest = "0.9.5"
```
Comment (omershlo) [inline gotham-client/Cargo.toml]: #28 ?

**Comment by omershlo** [comment]
I approve

**Comment by oleiba** [comment]
It seems that I can't merge without your "github" approval.
Can you please go to "Files changed" tab, "Review changes" and "approve"?

**Comment by burdges** [comment]
I'm curious: Why only two parties for a Schnorr signature?

**Comment by omershlo** [comment]
The Gotham city architecture includes a network stack. 
For simplicity purposes  we wanted to rely on a classical server<>client design where the client makes rest-api calls to the server to facilitate the protocol and so on.  This way we avoid p2p and still get distributed computation (although with just 2 parties)

**Pull request merged**